spec/requests/api/teams_spec.rb

```diff
@@ -2,13 +2,15 @@
 
 
 RSpec.describe "/teams", type: :request do
-
+  let!(:league) { create :league }
```

Comment:
You don't need to always use `let!` (with the `!` at the end). You can also use let (without the `!`) to lazy load the variable you're declaring. Check this small post to better understand what I'm trying to say.

https://mixandgo.com/learn/let-vs-instance-variables-in-rspec

Reply:
###  the test not pass

Erro:

Team validations is expected to validate that :name is case-sensitively unique
     Failure/Error: it { should validate_uniqueness_of(:name)}
     
     Shoulda::Matchers::ActiveRecord::ValidateUniquenessOfMatcher::ExistingRecordInvalid:
       validate_uniqueness_of works by matching a new record against an
       existing record. If there is no existing record, it will create one
       using the record you provide.
     
       While doing this, the following error was raised:
     
         PG::NotNullViolation: ERROR:  null value in column "league_id" of relation "teams" violates not-null constraint
```suggestion
  let!(:league) { create :league }
```
         DETAIL:  Failing row contains (124, null, null, 2021-06-16 14:10:01.251013, 2021-06-16 14:10:01.251013).
     
       The best way to fix this is to provide the matcher with a record where
       any required attributes are filled in with valid values beforehand.
     # ./spec/models/team_spec.rb:9:in `block (3 levels) in <top (required)>'
     # ------------------
     # --- Caused by: ---
     # PG::NotNullViolation:
     #   ERROR:  null value in column "league_id" of relation "teams" violates not-null constraint
     #   DETAIL:  Failing row contains (124, null, null, 2021-06-16 14:10:01.251013, 2021-06-16 14:10:01.251013).
     #   ./spec/models/team_spec.rb:9:in `block (3 levels) in <top (required)>'